test/release/examples/primers/forallLoops.chpl

```diff
@@ -0,0 +1,305 @@
+// Forall Loops
+
+/*
+This primer illustrates forall loops, which are a way to leverage
+data parallelism or engage :ref:`user-defined parallel iterators
+<primers-parIters>`. 
+
+Like serial for-loops, forall loops can iterate over a data structure,
+an iterator, or a zippered combination of these. Unlike for-loops,
+multiple iterations of a forall-loop can potentially execute in parallel.
+Parallelism is determined by the data structure or iterator being iterated
+over. For a zippered forall loop, parallelism is determined by the "leader",
+which is the first data structure or iterator in the zippered list.
+
+Chapel has forall statements and forall expressions. Each form has
+two varieties, "must-parallel" and "may-parallel".
+
+ - The must-parallel forms are written using the ``forall`` keyword.
+   They require that the leader iterable provide a parallel iterator.
+   Note that there are no requirements on the behavior of the parallel
+   iterator. For example, it can execute serially, in which case
+   the "must-parallel" loop that invokes it also executes serially.
+
+ - The may-parallel forms are written using brackets ``[ ]``.
+   They invoke the parallel iterator if the leader provides it,
+   otherwise fall back on the serial iterator.
+
+As with for-loops, the body of a forall statement is a statement or a
+a block statement, whereas the body of a forall expression is an expression.
+
+"Must-parallel" forall statement
+--------------------------------
+
+In the following example, the forall loop iterates over the array indices
+in parallel:
+*/
+
+config const n = 9;
+var A: [1..n] real;
+
+forall i in 1..n {
+  A[i] = i;
+}
+
+writeln("After setting up, A is:");
+writeln(A);
+writeln();
+
+/*
+If ``A`` were a distributed array, each loop iteration would be
+executed on the locale where the corresponding array element resides.
```

Comment:
There's actually no guarantee of this, though it's certainly our intention for distributions we provide.  So it might make sense to qualify this with "typically be executed" or "Block-distributed array" or ...?